docs/cpp/source/notes/tensor_cuda_stream.rst

```diff
@@ -0,0 +1,189 @@
+Tensor CUDA Stream API 
+======================
+
+A `CUDA Stream`_ is a linear sequence of execution that belongs to a specific device.
+Pytorch C++ API has defined a CUDAStream class and a few useful functions to support CUDA Stream operations.
+You can find them in `CUDAStream.h`_. This note provides more details on how to use Pytorch C++ CUDA Stream APIs.
+
+.. _CUDA Stream: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#streams
+.. _CUDAStream.h: https://pytorch.org/cppdocs/api/file_c10_cuda_CUDAStream.h.html#file-c10-cuda-cudastream-h
+.. _CUDAStreamGuard.h: https://pytorch.org/cppdocs/api/structc10_1_1cuda_1_1_c_u_d_a_stream_guard.html
+
+Get CUDA stream
+*****************
+
+Pytorch C++ API provides the following ways to get CUDA stream:
+
+1. Get a new stream from the CUDA stream pool, streams are preallocated from the pool and returned in a round-robin fashion.
+
+.. code-block:: cpp
+  
+  CUDAStream getStreamFromPool(const bool isHighPriority = false, DeviceIndex device = -1);
+
+.. tip::
+
+  You can request a stream from the high priority pool by setting isHighPriority to true, or a stream for a specific device
+  by setting device (defaulting to the current CUDA stream).
```

Comment:
maybe "by passing a device index" to be more consistent with the language below? 

By default I think this returns a low priority stream on the current device, but not necessarily the current CUDA stream. Or am I misremember how this works?

Reply:
This should be "defauting to the current CUDA stream's device index)